### PR TITLE
HoloPy Recipe

### PR DESCRIPTION
**Pull request by barkls**
Hello,

We would like to use conda-forge to distribute our package for analyzing holographic microscopy images. We rely on Fortran code, which I cannot figure out how to compile on Windows (see #1807), so we are settling for mac and linux builds for now. 

I'm looking forward to your comments!

Solomon Barkley

### CODE REVIEW

**Comment by barkls** [comment]
it looks like the CircleCI py34 test worked, but numpy failed to import in py35... I'm not sure if this is a problem with my recipe specifically?

**Comment by barkls** [comment]
Ok I've temporarily skipped python 3.5 just to get something passing. Let me know when you get a chance to take a look @conda-forge/staged-recipes
Thanks!

**Comment by barkls** [comment]
It looks like it wasn't the python 3.5 numpy import itself that was the problem, since it depends on whether or not a python 3.4 build already ran:

commit 57f750b        run py34 only                            passes
commit 7680989       run both py34 and py35            fails to import numpy for py35
commit 471ed5c       run py35 only                            passes

**Comment by barkls** [comment]
Still waiting on some comments/review of this recipe (not necessarily regarding the numpy import problem). @conda-forge/core

File: recipes/holopy/COPYING [hunk 1]
```diff
@@ -0,0 +1,203 @@
+ GNU GENERAL PUBLIC LICENSE
```
Comment (jakirkham) [inline recipes/holopy/COPYING]: Please drop this file. It is picked up from the downloaded source so we don't need it in the recipe.

File: recipes/holopy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/manoharan-lab/holopy.git
+  git_rev: 0fbb2f08181ed4f5e951ff880faaec62e5905a5f
+
+build:
+  number: 3
```
Comment (jakirkham) [inline recipes/holopy/meta.yaml]: Please reset to `0`.

File: recipes/holopy/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    - numpy
+    - scipy
+    - pyyaml
+    - pillow
+    - h5py
+    - emcee
+    - matplotlib
+    - xarray
+    - h5netcdf
+    - libgfortran  # [not win]
```
Comment (jakirkham) [inline recipes/holopy/meta.yaml]: Please drop as this is just part of `libgcc`.

File: recipes/holopy/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+build:
+  number: 3
+  skip: True  # [py2k or py34 or win]
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - numpy
```
Comment (jakirkham) [inline recipes/holopy/meta.yaml]: Does this link to NumPy? If so, we need `x.x` after `numpy`.

File: recipes/holopy/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    - setuptools
+    - gcc  # [not win]
+    - m2w64-gcc-fortran  # [win]
+
+  run:
+    - python
+    - numpy
```
Comment (jakirkham) [inline recipes/holopy/meta.yaml]: Same as above.

File: recipes/holopy/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    - pyyaml
+    - pillow
+    - h5py
+    - emcee
+    - matplotlib
+    - xarray
+    - h5netcdf
+    - libgfortran  # [not win]
+    - libgcc  # [not win]
+    - m2w64-gcc-libgfortran  # [win]
```
Comment (jakirkham) [inline recipes/holopy/meta.yaml]: Do we need `libgcc`?

File: recipes/holopy/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    It has tools for managing data and metadata, doing reconstructions, computing light scattering
+    from micron scale particles or clusters of such particles, and fitting scattering models to
+    digital holograms to measure particle parameters. HoloPy is free and open source software (GPL v3).
+  doc_url: https://readthedocs.org/projects/holopy/
+  dev_url: https://github.com/manoharan-lab/holopy
+
+extra:
+  recipe-maintainers:
+    - barkls
+    - vnmanoharan
```
Comment (jakirkham) [inline recipes/holopy/meta.yaml]: Are you ok with being added as a maintainer here, @vnmanoharan ?

File: recipes/holopy/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,63 @@
+{% set name = "HoloPy" %}
+{% set version = "3.0_beta" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/manoharan-lab/holopy.git
+  git_rev: 0fbb2f08181ed4f5e951ff880faaec62e5905a5f
```
Comment (jakirkham) [inline recipes/holopy/meta.yaml]: Is there a tarball that we can download? We prefer not to clone from source.

File: recipes/holopy/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,63 @@
+{% set name = "HoloPy" %}
+{% set version = "3.0_beta" %}
```
Comment (jakirkham) [inline recipes/holopy/meta.yaml]: Is there a non-beta release? Also this versioning style is a little odd.

File: recipes/holopy/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/manoharan-lab/holopy.git
+  git_rev: 0fbb2f08181ed4f5e951ff880faaec62e5905a5f
+
+build:
+  number: 3
+  skip: True  # [py2k or py34 or win]
+  script: python setup.py install --single-version-externally-managed --record record.txt
```
Comment (jakirkham) [inline recipes/holopy/meta.yaml]: Please drop this line. It is ignored when `build.sh` is used.

File: recipes/holopy/meta.yaml (hunk 7 above)
Comment (vnmanoharan) [inline recipes/holopy/meta.yaml]: Yes, I am.

**Comment by scopatz** [comment]
Any movement on this @barkls?

**Comment by barkls** [comment]
Yes, I have made some changes but haven't pushed them yet. 

I am not sure what to do about my skip line. See my comment replying to @jakirkham  above (I quote here):

> Thanks for the review.

Regarding windows skip: We would like to be able to support Windows, but I haven't been able to successfully compile our fortran code so far. You mentioned in an earlier conversation (#1807) that it might be possible with m2w64-gcc-fortran. We are skipping windows for now, but I kept the dependencies around because we hope to eventually build on windows.

Regarding python skip: We only support python 3. I was getting a failure with numpy imports during the py35 build on circle or travis that only occurred when the py34 build preceded it. Either py34 or py35 works fine individually (see my earlier comments on this PR). Adding the numpy x.x may solve this?

EDIT: with numpy x.x, the py34 np111 build fails.

**Comment by scopatz** [comment]
I think we might now have a fix for the numpy piece of this on master. If you could try mergeing/rebasing that would be great!

File: recipes/holopy/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/manoharan-lab/holopy.git
+  git_rev: 0fbb2f08181ed4f5e951ff880faaec62e5905a5f
+
+build:
+  number: 3
+  skip: True  # [py2k or py34 or win]
```
Comment (jakirkham) [inline recipes/holopy/meta.yaml]: Ok.

> Regarding python skip: We only support python 3. I was getting a failure with numpy imports during the py35 build on circle or travis that only occurred when the py34 build preceded it. Either py34 or py35 works fine individually (see my earlier comments on this PR). Adding the numpy x.x may solve this?
>
> with numpy x.x, the py34 np111 build fails.

As @scopatz mentioned below, I think we have worked around this issue in PR ( https://github.com/conda-forge/staged-recipes/pull/2126 ). Seems to be a `conda-build-all` bug.

**Comment by scopatz** [comment]
ping @barkls

**Pull request closed without merging**